CMakeLists.txt

```diff
@@ -380,7 +376,7 @@ if(UNITTEST)
 
     add_test(NAME test_CML_ldpcut
              COMMAND sh -c "cd ${CMAKE_CURRENT_SOURCE_DIR}/octave; SHORT_VERSION_FOR_CTEST=1 octave-cli -qf ldpcut.m")
-             set_tests_properties(test_CML_ldpcut PROPERTIES PASS_REGULAR_EXPRESSION "Nerr: 0")
+             set_tests_properties(test_CML_ldpcut PROPERTIES ENVIRONMENT "CMAKE_CML_PATH=${CMAKE_CURRENT_BINARY_DIR}/cml" PASS_REGULAR_EXPRESSION "Nerr: 0")
 
```

Comment:
Hmm, I wonder if we can make this "global" to all ctests somehow

Reply:
Some other alternatives:
1. Set the env variable in `COMMAND`
1. break the `set_tests_properties` into two, shorter lines.  The one big line approach obscures the test design a little, and is hard to read on small screens.

Reply:
Could we get `cmake` to create the `.octaverc` file in `codec2/octave` when building CML?  Then every time octave is started in `codec/octave` it would read this env variable.  This does kind of break the "out of source build" paradigm though :thinking:

Reply:
I consolidated all of the environment sets for the CML tests at the bottom of CMakeLists.txt and updated the Octave scripts to default to ~/cml/ (old behavior) if CMAKE_CML_PATH isn't specified. Hope that helps!

Reply:
@tmiw yes I think the former is an improvement, however I'm not sure about https://github.com/drowe67/codec2/pull/267/commits/fda9d500a6b12cdbf459f0aba6214b35d1039166.  This would imply we need to create another `~/cml` directory, when one has already been built in `$(CMAKE_CURRENT_BINARY_DIR}/cml`.

What do you think of the solution I have proposed [here](https://github.com/drowe67/codec2/pull/267#discussion_r767418195)?  I've typed up the instructions in [README.md](https://github.com/drowe67/codec2/pull/267/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5)

Can we pls discuss the way fwd before making any more changes?  There's no rush on this PR, and I'd prefer we reach a consensus.

Reply:
Creating ~/cml/ is optional for sure and is really just a fallback if CMAKE_CML_PATH isn't set somewhere (command line or .octaverc as you mentioned). Perhaps the variable can be renamed to indicate that it can be set outside of CMake as well?

Reply:
Yes renaming it is a good idea - how about we rename it to `CML_PATH`?

Would you mind if we reverted https://github.com/drowe67/codec2/commit/fda9d500a6b12cdbf459f0aba6214b35d1039166?  I quite liked your original solution ... and `~/cml` won't be present on any machines from now on.

Reply:
Would removing the length check in ldpc.m work as well? I'm fine with reverting if that's preferred, though.

Reply:
Yes now that I look at `init_cml()` I can see what you mean.  I agree - lets keep https://github.com/drowe67/codec2/commit/fda9d500a6b12cdbf459f0aba6214b35d1039166 and refactor `init_cml()`, it can be much simpler now.

I'm working on the PR now so if you like I can remove that length check later today.

Reply:
Go for it. I won't be able to touch it until tonight anyway.

Can you create a repo for CML? I can work on getting rid of the compiler warnings.

Reply:
https://github.com/drowe67/cml, it has a GitHub action workflow to build.  Will be create a release zip file from this repo that codec2 pulls when building?

Reply:
No need. CMake will clone the repo when building the unit tests.

Reply:
Oops I accidentally asked this question twice - deleted the 2nd comment/question!